rxjava2/src/main/java/io/vertx/reactivex/core/json/ObservableUnmarshaller.java

```diff
@@ -0,0 +1,78 @@
+package io.vertx.reactivex.core.json;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import io.reactivex.ObservableOperator;
+import io.reactivex.Observer;
+import io.reactivex.annotations.NonNull;
+import io.reactivex.disposables.Disposable;
+import io.vertx.core.buffer.Buffer;
+
+import java.io.IOException;
+
+import static io.vertx.core.json.Json.mapper;
+import static java.util.Objects.nonNull;
+
+/**
+ * An operator to unmarshall json to pojos.
+ *
+ * @author <a href="mailto:julien@julienviet.com">Julien Viet</a>
+ */
+public class ObservableUnmarshaller<T, B> implements ObservableOperator<T, B> {
```

Comment:
The snippet above is for RxJava 1. With RxJava2 it must be updated to deal with the empty buffer case. Actually, an empty `Observable` would be better, the user could still decide what to do with standard operators and behavior would be consistent with RxJava2 implementation